src/cbc_sdk/platform/policies.py

```diff
@@ -986,6 +1277,172 @@ def validate(self):
+
+        Required Permissions:
+            org.policies(UPDATE)
+        """
+        was_deleted = False
+        try:
+            self._parent._on_deleted_rule_config(self)
+            was_deleted = True
+        finally:
+            if was_deleted:
+                self._parent = None
+
+    def get_parameter(self, name):
+        """
+        Returns a parameter value from the rule configuration.
+
+        Args:
+            name (str): The parameter name.
+
+        Returns:
+            Any: The parameter value, or None if there is no value.
+        """
+        params = self._info['parameters']
+        return params.get(name, None)
+
+    def set_parameter(self, name, value):
+        """
+        Sets a parameter value into the rule configuration.
+
+        Args:
+            name (str): The parameter name.
+            value (Any): The new value to be set.
+        """
+        params = self._info['parameters']
+        old_value = params.get(name, None)
+        if old_value != value:
+            if 'parameters' not in self._dirty_attributes:
+                self._dirty_attributes['parameters'] = params
+                new_params = copy.deepcopy(params)
+            else:
+                new_params = params
+            new_params[name] = value
+            self._info['parameters'] = new_params
+
+    def validate(self):
+        """
+        Validates this rule configuration against its constraints.
+
+        Raises:
+            InvalidObjectError: If the rule object is not valid.
+        """
+        super(PolicyRuleConfig, self).validate()
+
+        if self._parent is not None:
+            # set high-level fields
+            valid_configs = self._parent.valid_rule_configs()
+            data = valid_configs.get(self._model_unique_id, {})
+            self._info.update(data)
+            if 'inherited_from' not in self._info:
+                self._info['inherited_from'] = 'psc:region'
+
+        # validate parameters
+        if self._parent is None:
+            parameter_validations = Policy.get_ruleconfig_parameter_schema_directly(self._cb, self._model_unique_id)
+        else:
+            parameter_validations = self._parent.get_ruleconfig_parameter_schema(self._model_unique_id)
+        my_parameters = self._info.get('parameters', {})
+        for k, v in parameter_validations.items():
+            if k in my_parameters:
+                if v['type'] == 'string':
+                    if not isinstance(my_parameters[k], str):
+                        raise InvalidObjectError(f"rule configuration parameter '{k}' is not a string")
+                    if ('enum' in v) and (my_parameters[k] not in v['enum']):
+                        raise InvalidObjectError(f"invalid value '{my_parameters[k]}' "
+                                                 f"for rule configuration parameter '{k}'")
+                else:
+                    raise ApiError(f"internal error: unknown parameter type {v['type']}")
```

Comment:
We shouldn't throw our own exceptions for errors in our own code? Besides, `validate` is already documented in the base class as throwing `InvalidObjectError`.

Reply:
We can continue to use `InvalidObjectError` as we have that pattern already though we need to enhance the error doc string as it doesn't match our current usage
```
class InvalidObjectError(ApiError):
    """An invalid object was received by the server."""
    pass
```

I don't approve of `ApiError` as that is not from the API. I understand it's our base class error but it's name is misleading in this case. Additionally, I am not sure the current condition makes sense as it currently prevents any other parameter types than string. What about integer or boolean as I know we use boolean for HBFW? This code needs to be more future proof for retrieving and verifying all or the common json parameter types

Reply:
It will incorporate other parameter types once I know what those look like in the rule config parameter schema that we receive from the server.  Right now, HBFW does not _have_ any rule config parameter schema that I can see, at least from the presentation data.

Reply:
As for the exception, what would you suggest here instead of `ApiError`?

Reply:
Plan is to replace with jsonschema validator

Reply:
I replaced the core of validation with the `jsonschema.validate` call. It works.